minemeld/ft/taxii2.py

```diff
@@ -0,0 +1,634 @@
+            LOG.error('%s - wrong initial_interval format: %s', self.name, self.initial_interval)
+            self.initial_interval = 86400
+        self.max_poll_dt = self.config.get('max_poll_dt', 86400)
+
+        # options for processing
+        self.ip_version_auto_detect = self.config.get('ip_version_auto_detect', True)
+        self.ignore_composition_operator = self.config.get('ignore_composition_operator', False)
+        self.create_fake_indicator = self.config.get('create_fake_indicator', False)
+        self.hash_priority = self.config.get('hash_priority', _STIX2_MINEMELD_HASHES)
+        self.lower_timestamp_precision = self.config.get('lower_timestamp_precision', False)
+
+        self.auth_type = self.config.get('auth_type', 'none')
+
+        self.username = self.config.get('username', None)
+        self.password = self.config.get('password', None)
+
+        self.api_key = self.config.get('api_key', None)
+
+        self.discovery_service = self.config.get('discovery_service', None)
+        self.api_root = self.config.get('api_root', None)
+        self.collection = self.config.get('collection', None)
+        self.verify_cert = self.config.get('verify_cert', True)
+
+        self.enabled = self.config.get('enabled', 'no')
+
+        self.client = None
+        self.taxii_collection = None
+        self.taxii_version = None
+
+        self.side_config_path = self.config.get('side_config', None)
+        if self.side_config_path is None:
+            self.side_config_path = os.path.join(
+                os.environ['MM_CONFIG_DIR'],
+                '%s_side_config.yml' % self.name
+            )
+
+        # self.prefix = self.config.get('prefix', self.name)
+
+        # self.confidence_map = self.config.get('confidence_map', {'low': 40, 'medium': 60, 'high': 80})
+
+        self._load_side_config()
+
+    def _load_side_config(self):
+        try:
+            with open(self.side_config_path, 'r') as f:
+                sconfig = yaml.safe_load(f)
+
+        except Exception as e:
+            LOG.error('%s - Error loading side config: %s', self.name, str(e))
+            return
+
+        auth_type = sconfig.get('auth_type', None)
+        username = sconfig.get('username', None)
+        password = sconfig.get('password', None)
+        api_key = sconfig.get('api_key', None)
+
+        if api_key is not None:
+            self.api_key = api_key
+            LOG.info('{} - Loaded credentials from side config'.format(self.name))
+        elif username is not None and password is not None:
+            self.username = username
+            self.password = password
+            LOG.info('{} - Loaded credentials from side config'.format(self.name))
+
+        discovery_service = sconfig.get('discovery_service', None)
+        api_root = sconfig.get('api_root', None)
+        collection = sconfig.get('collection', None)
+        verify_cert = sconfig.get('verify_cert', None)
+
+        enabled = sconfig.get('enabled', None)
```

Comment:
This is a nice touch :-)

Reply:
Thanks! We thought the end users would like a chance to configure authentication before querying the server.